### PR TITLE
Install GCC in unit test container

### PR DESCRIPTION
**Pull request by 14rcole**

### CODE REVIEW

**Comment by samvarankashyap** [comment]
@14rcole : Please fix the indentation. 
Furthermore, If gcc is not available there is a high chance that python-devel might not be available too. 
Could you add that too ?

**Comment by p3ck** [comment]
we may have to re-think the way the unittests are running.

**Comment by 14rcole** [comment]
@p3ck is there any particular reason that we use the `continuous-infra/jenkins` container to run the unit tests?  Do we really need jenkins for anything?

**Comment by samvarankashyap** [comment]
@p3ck Is there any way we could run it inside a custom container rather than a Jenkins container ??

**Comment by p3ck** [comment]
> @p3ck Is there any way we could run it inside a custom container rather than a Jenkins container ??

HDSL uses a tool called linchpin to do it's provisioning.  If that supports provisioning containers then yes. ;)

**Comment by 14rcole** [comment]
@samvarankashyap The container is defined here, we can just change that
https://github.com/CentOS-PaaS-SIG/linchpin/blob/develop/config/Dockerfiles/JenkinsfileUnitTests#L5

*but* it appears to be pulling from an openshift registry somewhere so we'll have to see what images are available in that registry

**Comment by samvarankashyap** [comment]
[test]

File: playbooks/tests/unit_tests.yml [hunk 1]
```diff
@@ -3,7 +3,16 @@
...
-          - name: Install Test Reqs
+          - name: Install yum Test Reqs
+            yum:
+                    name: "{{ packages }}"
+                    state: present
+            vars:
+                    packages:
+                          - gcc
+                          - python-devel
+            become: yes
```
Comment (samvarankashyap) [inline playbooks/tests/unit_tests.yml]: @14rcole I dont think you need sudo here if we are running as root user in the container

File: config/Dockerfiles/JenkinsfileUnitTests [hunk 2]
```diff
@@ -1,9 +1,9 @@
 // Define the openshift pod name, docker repo url, namespace, and service acct
 // for the DSL pod template.
 dslPodName = "contraDsl-${UUID.randomUUID()}"
-dockerRepoURL = env.DOCKER_REPO_URL ?: '172.30.254.79:5000'
-openshiftNamespace = env.OPENSHIFT_NAMESPACE ?: 'continuous-infra'
-openshiftServiceAccount = env.OPENSHIFT_SERVICE_ACCOUNT ?: 'jenkins'
+dockerRepoURL = env.DOCKER_REPO_URL ?: ''
+openshiftNamespace = env.OPENSHIFT_NAMESPACE ?: 'docker.io'
+openshiftServiceAccount = env.OPENSHIFT_SERVICE_ACCOUNT ?: 'fedora:29'
```
Comment (samvarankashyap) [inline config/Dockerfiles/JenkinsfileUnitTests]: @14rcole Is there a service account named fedora:29 ???

Comment (samvarankashyap) [reply config/Dockerfiles/JenkinsfileUnitTests]: @14rcole Further there is no namespace named docker.io , it should be continous-infra and docker repo url would be docker.io i guess

Comment (14rcole) [reply config/Dockerfiles/JenkinsfileUnitTests]: The environment variables are poorly named, but we end up doing a pull of `docker://dockerRepoURL/openshiftNamespace/openshiftServiceAccount` e.g. `docker://172.35.15.12/continuous-infra/jenkins:latest`

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
The solution ended up being to remove pycrypto.  It's the only package that requires a c compiler but we don't seem to use it anywhere

**Comment by samvarankashyap** [comment]
@14rcole pycrypto used to be requirement for ansible 2.0 to 2.3 it was replaced by pycryptodome.
I don't think we need it anymore.

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by samvarankashyap** [comment]
@14rcole you may merge it

**Comment by 14rcole** [comment]
[merge]

**Pull request merged**